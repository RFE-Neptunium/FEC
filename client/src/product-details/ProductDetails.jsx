/* eslint-disable radix */
/* eslint-disable react/prop-types */
import React from 'react';

function ProductDetails({ product, currentStyle, productReviews }) {
  if (Object.keys(productReviews.ratings).length > 0) {
    let starRating = '*';

    let averageRating = 0;
    let totalRatings = 0;
    for (let i = 1; i <= 5; i += 1) {
      averageRating += parseInt(productReviews.ratings[i]) * i;
      totalRatings += parseInt(productReviews.ratings[i]);
    }

    averageRating /= totalRatings;
    averageRating = averageRating.toFixed(2);

    if (currentStyle.sales_price) {
      console.log('1');
      return (
        <div className="productDetails">
          <p id="category">{product.category}</p>
          <h1 id="title">{product.name}</h1>
          <h2 id="rating">{`${averageRating} stars. Read all ${totalRatings} review(s)`}</h2>
          <div className="saleOutlay">
            <h3 id="nonPrice">{'$' + currentStyle.original_price}</h3>
            <h3 id="salesPrice">{currentStyle.sales_price}</h3>
          </div>
          <p id="overview">{product.description}</p>
          <button id="Facebook" type="button">Facebook</button>
          <button id="Twitter" type="button">Twitter</button>
          <button id="Pinterest" type="button">Pinterest</button>
        </div>
      );
    }
    console.log('2');
    return (
      <div className="productDetails">
        <p id="category">{product.category}</p>
        <h1 id="title">{product.name}</h1>
        <h2 id="rating">{`${averageRating} stars. Read all ${totalRatings} review(s)`}</h2>
        <h3 id="price">{'$' + currentStyle.original_price}</h3>
        <p id="overview">{product.description}</p>
        <button id="Facebook" type="button">Facebook</button>
        <button id="Twitter" type="button">Twitter</button>
        <button id="Pinterest" type="button">Pinterest</button>
      </div>
    );
  }

  if (currentStyle.sales_price) {
    console.log('3');
    return (
      <div className="productDetails">
        <p id="category">{product.category}</p>
        <h1 id="title">{product.name}</h1>
        <div className="saleOutlay">
          <h3 id="nonPrice">{'$' + currentStyle.original_price}</h3>
          <h3 id="salesPrice">{currentStyle.sales_price}</h3>
        </div>
        <p id="overview">{product.description}</p>
        <button id="Facebook" type="button">Facebook</button>
        <button id="Twitter" type="button">Twitter</button>
        <button id="Pinterest" type="button">Pinterest</button>
      </div>
    );
  }

  return (
    <div className="productDetails">
      <p id="category">{product.category}</p>
      <h1 id="title">{product.name}</h1>
      <h3 id="price">{'$' + currentStyle.original_price}</h3>
      <p id="overview">{product.description}</p>
      <button id="Facebook" type="button">Facebook</button>
      <button id="Twitter" type="button">Twitter</button>
      <button id="Pinterest" type="button">Pinterest</button>
    </div>
  );
}

export default ProductDetails;
