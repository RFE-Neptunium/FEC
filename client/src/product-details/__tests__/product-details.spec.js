/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import ProductDetails from '../ProductDetails';

describe('currentStyle', () => {
  test('it should have a length of one', () => {
    const currentStyle = {
      title: 'Placeholder Title 1',
      category: 'Placeholder Category 1',
      rating: 3.5,
      price: '$500.00',
      overview: 'Placeholder Overview 1',
      reviewNum: 4,
      stock: { small: 5, medium: 3, large: 2 },
      gallery: [],
      currentSize: 'Choose a Size',
      currentQuantity: 0,
    };

    const keys = Object.keys(currentStyle);

    expect(keys.length).toEqual(10);
  });
});

it('should have a type of objecy', () => {
  const component = <ProductDetails />;

  expect(typeof component).toEqual('object');
});
