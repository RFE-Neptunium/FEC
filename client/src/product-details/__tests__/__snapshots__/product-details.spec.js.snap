// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render tests AddToCart renders correctly 1 1`] = `
<div
  className="outOfStock"
>
  <select
    className="sizeOptions"
    isDisabled={true}
  >
    <option>
      OUT OF STOCK
    </option>
  </select>
  <select
    className="amountOptions"
  >
    <option
      value="-"
    >
      -
    </option>
  </select>
</div>
`;

exports[`render tests AddToCart renders correctly 2 1`] = `
<div
  className="addToCart"
>
  <p
    className="sizeAlert"
  />
  <form>
    <select
      className="sizeOptions"
      onChange={[Function]}
    >
      <option
        value="Select Size"
      >
        Select Size
      </option>
      <option
        value="M"
      >
        M
      </option>
    </select>
    <select
      className="amountOptions"
      onChange={[Function]}
    >
      <option
        value="-"
      >
        -
      </option>
    </select>
    <button
      className="cartButton"
      onClick={[Function]}
      type="button"
    >
      Add to Cart
    </button>
  </form>
</div>
`;

exports[`render tests AddToCart renders correctly 3 1`] = `
<div
  className="addToCart"
>
  <p
    className="sizeAlert"
  />
  <form>
    <select
      className="sizeOptions"
      onChange={[Function]}
    >
      <option
        value="Select Size"
      >
        Select Size
      </option>
      <option
        value="M"
      >
        M
      </option>
    </select>
    <select
      className="amountOptions"
      onChange={[Function]}
    >
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
      <option
        value={3}
      >
        3
      </option>
      <option
        value={4}
      >
        4
      </option>
      <option
        value={5}
      >
        5
      </option>
    </select>
    <button
      className="cartButton"
      onClick={[Function]}
      type="button"
    >
      Add to Cart
    </button>
  </form>
</div>
`;

exports[`render tests AddToCart renders correctly 4 1`] = `
<div
  className="addToCart"
>
  <p
    className="sizeAlert"
  />
  <form>
    <select
      className="sizeOptions"
      onChange={[Function]}
    >
      <option
        value="Select Size"
      >
        Select Size
      </option>
      <option
        value="M"
      >
        M
      </option>
    </select>
    <select
      className="amountOptions"
      onChange={[Function]}
    >
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
      <option
        value={3}
      >
        3
      </option>
      <option
        value={4}
      >
        4
      </option>
      <option
        value={5}
      >
        5
      </option>
      <option
        value={6}
      >
        6
      </option>
      <option
        value={7}
      >
        7
      </option>
      <option
        value={8}
      >
        8
      </option>
      <option
        value={9}
      >
        9
      </option>
      <option
        value={10}
      >
        10
      </option>
      <option
        value={11}
      >
        11
      </option>
      <option
        value={12}
      >
        12
      </option>
      <option
        value={13}
      >
        13
      </option>
      <option
        value={14}
      >
        14
      </option>
      <option
        value={15}
      >
        15
      </option>
    </select>
    <button
      className="cartButton"
      onClick={[Function]}
      type="button"
    >
      Add to Cart
    </button>
  </form>
</div>
`;

exports[`render tests ImageGallery renders correctly 1 1`] = `
<div
  className="imagegallery"
>
  <div
    className="thumbnailView"
  >
    <div
      className="carousel"
    >
      <img
        alt={0}
        onClick={[Function]}
        style={
          Object {
            "opacity": 0.7,
          }
        }
      />
    </div>
  </div>
  <div
    className="mainImageContainer"
  >
    <img
      className="currentImage"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src="./assets/checkmark.png"
      style={
        Object {
          "cursor": "auto",
        }
      }
    />
  </div>
</div>
`;

exports[`render tests ImageGallery renders correctly 2 1`] = `
<div
  className="imageGalleryModal"
>
  <div
    className="closeImageGalleryModal"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>
  <div
    className="thumbnailView"
  >
    <div
      className="carousel"
    >
      <img
        alt={0}
        onClick={[Function]}
        style={
          Object {
            "opacity": 0.7,
          }
        }
      />
    </div>
  </div>
  <div
    className="mainImageContainerModal"
    onClick={[Function]}
    onMouseMove={[Function]}
    style={
      Object {
        "transform": "scale(1)",
        "transformOrigin": "center",
      }
    }
  >
    <img
      className="currentImage"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src="./assets/checkmark.png"
      style={
        Object {
          "cursor": "zoom-in",
        }
      }
    />
  </div>
</div>
`;

exports[`render tests ImageGallery renders correctly 3 1`] = `
<div
  className="imagegallery"
>
  <div
    className="thumbnailView"
  >
    <div
      className="carousel"
    >
      <img
        alt={0}
        onClick={[Function]}
        style={
          Object {
            "opacity": 0.7,
          }
        }
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={1}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={2}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={3}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={4}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={5}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={6}
        onClick={[Function]}
      />
    </div>
    <div
      className="imageGalleryNext"
      onClick={[Function]}
    >
      〉
    </div>
  </div>
  <div
    className="mainImageContainer"
  >
    <img
      className="currentImage"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src="./assets/checkmark.png"
      style={
        Object {
          "cursor": "auto",
        }
      }
    />
  </div>
</div>
`;

exports[`render tests ImageGallery renders correctly 4 1`] = `
<div
  className="imagegallery"
>
  <div
    className="thumbnailView"
  >
    <div
      className="imageGalleryPrev"
      onClick={[Function]}
    >
      〈
    </div>
    <div
      className="carousel"
    >
      <img
        alt={1}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={2}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={3}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={4}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={5}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={6}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={7}
        onClick={[Function]}
      />
    </div>
    <div
      className="imageGalleryNext"
      onClick={[Function]}
    >
      〉
    </div>
  </div>
  <div
    className="mainImageContainer"
  >
    <img
      className="currentImage"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src="./assets/checkmark.png"
      style={
        Object {
          "cursor": "auto",
        }
      }
    />
  </div>
</div>
`;

exports[`render tests ImageGallery renders correctly 5 1`] = `
<div
  className="imagegallery"
>
  <div
    className="thumbnailView"
  >
    <div
      className="imageGalleryPrev"
      onClick={[Function]}
    >
      〈
    </div>
    <div
      className="carousel"
    >
      <img
        alt={1}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={2}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={3}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={4}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={5}
        onClick={[Function]}
      />
    </div>
    <div
      className="carousel"
    >
      <img
        alt={6}
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="mainImageContainer"
  >
    <img
      className="currentImage"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src="./assets/checkmark.png"
      style={
        Object {
          "cursor": "zoom-in",
        }
      }
    />
  </div>
</div>
`;

exports[`render tests Modal renders correctly 1 1`] = `
<div
  className="imageGalleryModal"
>
  <div
    className="closeImageGalleryModal"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>
  <div
    className="thumbnailView"
  >
    <div />
  </div>
  <div
    className="mainImageContainerModal"
    onClick={[Function]}
    onMouseMove={[Function]}
    style={
      Object {
        "transform": "scale(1)",
        "transformOrigin": "center",
      }
    }
  >
    <div />
  </div>
</div>
`;

exports[`render tests Modal renders correctly 2 1`] = `
<div
  className="imageGalleryModal"
>
  <div
    className="closeImageGalleryModal"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>
  <div
    className="thumbnailView"
  >
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div
      className="imageGalleryNext"
    >
      〉
    </div>
  </div>
  <div
    className="mainImageContainerModal"
    onClick={[Function]}
    onMouseMove={[Function]}
    style={
      Object {
        "transform": "scale(1)",
        "transformOrigin": "center",
      }
    }
  >
    <div />
  </div>
</div>
`;

exports[`render tests Modal renders correctly 3 1`] = `
<div
  className="imageGalleryModal"
>
  <div
    className="closeImageGalleryModal"
    onClick={[Function]}
  >
    <button
      type="button"
    >
      X
    </button>
  </div>
  <div
    className="thumbnailView"
  >
    <div
      className="imageGalleryPrev"
    >
      〈
    </div>
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div
      className="imageGalleryNext"
    >
      〉
    </div>
  </div>
  <div
    className="mainImageContainerModal"
    onClick={[Function]}
    onMouseMove={[Function]}
    style={
      Object {
        "transform": "scale(1)",
        "transformOrigin": "center",
      }
    }
  >
    <div />
  </div>
</div>
`;

exports[`render tests Modal renders correctly 4 1`] = `
<div
  className="imageGalleryModal"
>
  <div
    className="closeImageGalleryModal"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>
  <div
    className="thumbnailView"
  >
    <div
      className="imageGalleryPrev"
    >
      〈
    </div>
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
  <div
    className="mainImageContainerModal"
    onClick={[Function]}
    onMouseMove={[Function]}
    style={
      Object {
        "transform": "scale(1)",
        "transformOrigin": "center",
      }
    }
  >
    <div />
  </div>
</div>
`;

exports[`render tests Modal renders correctly 5 1`] = `
<div
  className="imageGalleryModal"
>
  <div
    className="closeImageGalleryModal"
    onClick={[Function]}
  >
    <button
      type="button"
    >
      X
    </button>
  </div>
  <div
    className="thumbnailView"
  >
    <div
      className="imageGalleryPrev"
    >
      〈
    </div>
    <div />
    <div />
    <div />
    <div />
    <div />
    <div
      className="imageGalleryNext"
    >
      〉
    </div>
  </div>
  <div
    className="mainImageContainerModal"
    onClick={[Function]}
    onMouseMove={[Function]}
    style={
      Object {
        "transform": "scale(2.5)",
        "transformOrigin": "0px 0px",
      }
    }
  >
    <div />
  </div>
</div>
`;

exports[`render tests Overview renders correctly 1`] = `null`;

exports[`render tests ProductDetails renders correctly 1 1`] = `
<div
  className="productDetails"
>
  <p
    id="category"
  />
  <h1
    id="title"
  />
  <div
    id="rating"
  >
    <div
      className="starRating"
    >
      <div
        className="sc-bczRLJ bSgFhy"
      >
        <div
          className="sc-gsnTZi bRVXLD"
        >
          <div
            className="sc-dkzDqf gGtFEz"
            width="NaN%"
          />
        </div>
      </div>
    </div>
    <div
      onClick={[Function]}
    >
      <h3
        className="totalRatings"
      >
        Read all 
        NaN
         review(s)
      </h3>
    </div>
  </div>
  <h3
    id="price"
  >
    $undefined
  </h3>
  <p
    id="overview"
  />
</div>
`;

exports[`render tests ProductDetails renders correctly 2 1`] = `
<div
  className="productDetails"
>
  <p
    id="category"
  />
  <h1
    id="title"
  />
  <h3
    id="price"
  >
    $undefined
  </h3>
  <p
    id="overview"
  />
</div>
`;

exports[`render tests ProductDetails renders correctly 3 1`] = `
<div
  className="productDetails"
>
  <p
    id="category"
  />
  <h1
    id="title"
  />
  <div
    id="rating"
  >
    <div
      className="starRating"
    >
      <div
        className="sc-bczRLJ bSgFhy"
      >
        <div
          className="sc-gsnTZi bRVXLD"
        >
          <div
            className="sc-dkzDqf gGtFEz"
            width="NaN%"
          />
        </div>
      </div>
    </div>
    <div
      onClick={[Function]}
    >
      <h3
        className="totalRatings"
      >
        Read all 
        NaN
         review(s)
      </h3>
    </div>
  </div>
  <div
    className="saleOutlay"
  >
    <h3
      id="nonPrice"
    >
      $undefined
    </h3>
    <h3
      id="salesPrice"
    >
      5
    </h3>
  </div>
  <p
    id="overview"
  />
</div>
`;

exports[`render tests ProductDetails renders correctly 4 1`] = `
<div
  className="productDetails"
>
  <p
    id="category"
  />
  <h1
    id="title"
  />
  <div
    className="saleOutlay"
  >
    <h3
      id="nonPrice"
    >
      $undefined
    </h3>
    <h3
      id="salesPrice"
    >
      5
    </h3>
  </div>
  <p
    id="overview"
  />
</div>
`;

exports[`render tests StyleSelector renders correctly 1 1`] = `
<div
  className="styleSelector"
>
  <p
    className="currentStyleSelector"
  />
</div>
`;

exports[`render tests StyleSelector renders correctly 2 1`] = `
<div
  className="styleSelector"
>
  <p
    className="currentStyleSelector"
  >
    exampleName
  </p>
  <div
    className="styleOption"
  >
    <img
      alt="1"
      onClick={[Function]}
      src={Object {}}
    />
  </div>
</div>
`;

exports[`render tests StyleSelector renders correctly 3 1`] = `
<div
  className="styleSelector"
>
  <p
    className="currentStyleSelector"
  >
    exampleName
  </p>
  <div
    className="styleOption"
  >
    <img
      alt="1"
      onClick={[Function]}
      src={Object {}}
    />
  </div>
</div>
`;
